### PR TITLE
feat: support arbitrary APPNAME and correct GPIOs

### LINKED ISSUE #56: Support for non-Nebra APPNAME
Currently in the main code we have:

https://github.com/NebraLtd/hm-config/blob/marvinmarnold/refactor/src/main.py#L76-L87

This pulls the `APPNAME` parameter from the [variant definitions file](https://github.com/NebraLtd/helium-hardware-definitions/blob/master/variant_definitions.py).

However, some of the variants (non-Nebra and DIY) do not have an `APPNAME` which causes this to fail.

My suggestion would be to handle this by having a check for `APPNAME` and if there is none, then the `localName` and `variant` variables would then become based on the `FRIENDLY` name.

Something like:

```python
class ConfigAdvertisement(Advertisement):
    # BLE advertisement
    def __init__(self, index):
        global variantDetails
        Advertisement.__init__(self, index, "peripheral")
        if 'APPNAME' in variantDetails:
            variant = variantDetails['APPNAME']
            macAddr = open("/sys/class/net/eth0/address").readline()\
                .strip().replace(":", "")[-6:].upper()
            localName = "Nebra %s Hotspot %s" % (variant, macAddr)
        else:
            variant = variantDetails['FRIENDLY']
            macAddr = open("/sys/class/net/eth0/address").readline()\
                .strip().replace(":", "")[-6:].upper()
            localName = "%s %s" % (variant, macAddr)
        self.add_local_name(localName)
        self.include_tx_power = True
        self.service_uuids = ["0fda92b2-44a2-4af2-84f5-fa682baa2b8d"]
```

### LINKED ISSUE #58: Correct outdoor GPIOs
When we pull in the variant definitions to lib in the new PR that takes over from #54 we need to make sure to update the outdoor units to the correct GPIOs.

See here for variant file https://github.com/NebraLtd/hm-config/blob/marvinmarnold/refactor/lib/variant_definitions.py

Relates to #57 and https://github.com/NebraLtd/hm-pktfwd/pull/31 and https://github.com/NebraLtd/helium-hardware-definitions/commit/577ce0ebca4349398480e01842aee4ad2662d2d4

FYI @marvinmarnold

### PR DESCRIPTION
**Pull request by marvinmarnold**
**Why**
When advertising, we pull the APPNAME parameter from the variant definitions file. However, some of the variants (non-Nebra and DIY) do not have an APPNAME which causes this to fail.

Also, different variants use different GPIOs, which need to be correctly used.

**How**
- renamed test file for consistency with underlying class
- added hardware definitions dependency and load it in app
- updated naming logic based on variant details
- get rid of isIndoorVariant helper
- add Mock.GPIO test dependency
- fix an unrelated issue with setting miner_keys when they are
    malformed, discovered during testing.
- having difficulty testing gatewayconfig_app because state is
    not being properly cleared between tests

**References**
Closes #56 
Closes #58

### CODE REVIEW

**Comment by shawaj** [comment]
Working well on a RAK based hotspot and light Hotspot

Check small-smoke on helium-light-test

File: gatewayconfig/gatewayconfig_app.py [hunk 1]
```diff
@@ -71,16 +73,9 @@ def init_nmcli(self):
...
     def init_gpio(self):
-        if is_indoor_variant(self.variant):
-            user_button_gpio = INDOOR_USER_BUTTON_GPIO
-            status_led_gpio = INDOOR_STATUS_LED_GPIO
-        else:
-            user_button_gpio = OTHER_USER_BUTTON_GPIO
-            status_led_gpio = OTHER_STATUS_LED_GPIO
-        
-        self.user_button = Button(user_button_gpio, hold_time=USER_BUTTON_HOLD_SECONDS)
+        self.user_button = Button(self.get_button_pin(), hold_time=USER_BUTTON_HOLD_SECONDS)
```
Comment (marvinmarnold) [inline gatewayconfig/gatewayconfig_app.py:76]: `get_button_pin` created to make unit testing easier

File: gatewayconfig/gatewayconfig_app.py [hunk 2]
```diff
@@ -60,7 +62,7 @@ def stop(self):
         logger.debug("Stopping ConfigApp")
         GPIO.cleanup()
         # Quits the cputemp application
-        self.bluetooth_processor.quit()
+        self.bluetooth_services_processor.quit()
```
Comment (marvinmarnold) [inline gatewayconfig/gatewayconfig_app.py:65]: This is an unrelated bug discovered during unit testing

File: gatewayconfig/file_loader.py [hunk 3]
```diff
@@ -35,7 +35,10 @@ def read_miner_keys(miner_keys_filepath):
                 for key in json_line.keys():
                     miner_keys[key] = json_line[key]
             except json.JSONDecodeError:
-                miner_keys[key] = False
+                logger.error("Unable to correctly decode miner_keys. The app will continue running but may not behave as expected.")
+                miner_keys['pubkey'] = False
```
Comment (marvinmarnold) [inline gatewayconfig/file_loader.py:39]: We should talk about this behavior more. The current logic will allow the application to start-up if the miner_keys file exists, even if it is malformed. Later on the app will choke when it tries to use those keys. Should we instead exit immediately when malformed JSON? Exponential backoff?

**Comment by marvinmarnold** [comment]
@vpetersson can we merge this?

File: gatewayconfig/file_loader.py (hunk 3 above)
Comment (shawaj) [reply gatewayconfig/file_loader.py:39]: 👍

**Review by vpetersson — APPROVED** [review]
I've only read over it and it looks reasonable. Needs to be tested on TestNet.

**Comment by shawaj** [comment]
@vpetersson @marvinmarnold pushed to testnet for testing https://github.com/NebraLtd/helium-miner-software/commit/2ef36bd19bf33dc482d740794d11c81a24bd86a5

**Comment by shawaj** [comment]
Seems to be working fine for me
